firmware/boron-ins-fsm/test/mocks/mock_ble.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Mock implementation for BLE functions and classes
+ */
+
+#pragma once
+
+#define BLE_GAP_ADV_SET_DATA_SIZE_MAX   (31)   //< Maximum data length for an advertising set.
+#define BLE_MAX_ADV_DATA_LEN            BLE_GAP_ADV_SET_DATA_SIZE_MAX
+
+#include "../inc/spark_wiring_vector.h"
+
+// Defines fake BleAdvertisingDataType values
+enum class BleAdvertisingDataType : uint8_t {
+    MANUFACTURER_SPECIFIC_DATA
+};
+
+// Fake class for BleAddress object
+class BleAddress {
+public:
+    BleAddress();
+
+    const BleAddress get() const {
+        return *this;
+    }
+};
+
+// Fake class for BleAdvertisingData object
+class BleAdvertisingData {
+public:
+    const BleAdvertisingData get(BleAdvertisingDataType bleAdvertisingDataType, uint8_t* advertisingData, int maxLength) const {
+        return *this;
+    }
+};
+
+// Fake class for BleScanResult object
+class BleScanResult
+{
+private:
+    BleAddress address_;
+    BleAdvertisingData advertisingData_;
+    BleAdvertisingData scanResponse_;
+    int8_t rssi_;
+
+public:
+    const BleAddress address() const {;
+        return address_;
+    }
+
+    const BleAdvertisingData advertisingData() const {
+        return advertisingData_;
+    }
+};
+
+// Fake class for BleScanFilter
+class BleScanFilter
+{
+private:
+    char* deviceNames_;
+    char* addresses_;
```

Comment:
Do these need to be defined here? I don't see them used anywhere.

Reply:
I left them in there since they are a part of the actual class and in case methods are added in the future that use them - I would be fine with removing them, though

Reply:
No, it's OK, you're right. It might come in handy later or be confusing if they aren't there. Thanks!